yt/analysis_modules/level_sets/clump_handling.py

```diff
@@ -302,14 +302,18 @@ def save_as_dataset(self, filename=None, fields=None):
             field_data = {}
             need_grid_positions = False
             for f in self.base.data._determine_fields(fields) + contour_fields:
-                field_data[f] = self.base[f]
                 if ds.field_info[f].particle_type:
                     if f[0] not in ptypes:
                         ptypes.append(f[0])
                     ftypes[f] = f[0]
                 else:
                     need_grid_positions = True
+                    if f[1] in 'xyz':
```

Comment:
The `dx`, `dy`, and `dz` fields will also be added automatically.  It's a small corner case, but probably best to make sure to skip over those, too.